docs/markdown/IDE-integration.md

```diff
@@ -219,7 +220,9 @@ in the `meson.build` (all dependencies are required by default). The
 inside a conditional block. In a real meson run these dependencies might not be
 used, thus they _may_ not be required, even if the `required` key is set. The
 `has_fallback` key just indicates whether a fallback was directly set in the
-`dependency()` function.
+`dependency()` function. The `version` key always contains the required version
+from the `meson.build` and **not** the actual version of the dependency on disc.
+The version is `null` if no version was specified in the `meson.build`.
```

Comment:
Version can be an array of multiple different strings [as per the docs](https://mesonbuild.com/Reference-manual.html#dependency). Thus this should be an array and if no version info was requested it should be the empty array, not `nul`. Also, `version` is not a very informative name on its own. Maybe it should be `requested_version` instead?

Reply:
Good catch. I have changed the `version` to be a list of strings now. However, I am not quite sure about renaming it to something like `requested_version`, `version_requirements`, etc. All the names I could come up with feel somehow bloated. Also, it should be pretty clear what is meant with `version` since the value is a list and almost all of the version strings will start with `>=` (in practice).

So I have kept it as `version` for now. I don't have strong feelings regarding this either way, but I think `version` (or `versions` alternatively?) might be the better choice here.